### PR TITLE
v2: Add encoding tests in rhp

### PR DESCRIPTION
**Pull request by chris124567**
Adds encoding roundtrip style test that brings coverage in rpc.go up to 72% and settings.go to 97%.

### CODE REVIEW

File: net/rhp/rpc_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		e := types.NewEncoder(&buf)
+		randVal, ok := quick.Value(reflect.TypeOf(val), rand.New(rand.NewSource(0)))
+		if ok {
+			randVal.Interface().(types.EncoderTo).EncodeTo(e)
+		} else {
+			val.EncodeTo(e)
+		}
+		e.Flush()
+
+		decval := reflect.ValueOf(val)
```
Comment (lukechampine) [inline net/rhp/rpc_test.go:80]: I worry about this aliasing pointers; to be safe, we should use `reflect.New(reflect.TypeOf(val))`

File: net/rhp/rpc_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	}
+
+	for _, val := range tests {
+		var buf bytes.Buffer
+		e := types.NewEncoder(&buf)
+		randVal, ok := quick.Value(reflect.TypeOf(val), rand.New(rand.NewSource(0)))
+		if ok {
+			randVal.Interface().(types.EncoderTo).EncodeTo(e)
+		} else {
+			val.EncodeTo(e)
```
Comment (lukechampine) [inline net/rhp/rpc_test.go:76]: does this mean that if `quick.Value` can't generate a value, we just use a zero-ed value instead?

I'd prefer to `t.Error` here. We should be generating values for all types.

File: net/rhp/rpc_test.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	"bytes"
+	"math/rand"
+	"reflect"
+	"testing"
+	"testing/quick"
+
+	"go.sia.tech/core/types"
+)
+
+func (p *HostSettings) Generate(rand *rand.Rand, size int) reflect.Value {
```
Comment (lukechampine) [inline net/rhp/rpc_test.go:13]: `// Generate implements quick.Generator.`

also, you can use `func (*HostSettings)` when the receiver variable is unused.

**Comment by lukechampine** [comment]
Looks like this caught some encoding bugs, nice 👌

I'm gonna start cracking down on commit hygiene -- pls squash this into one commit, and prefix the commit message with `net/rhp`

**Comment by chris124567** [comment]
Should be good to go.

**Pull request merged**